sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -0,0 +1,535 @@
+ * and the collection name. The following example illustrates various options for configuring the
+ * source:</p>
+ *
+ * <pre>{@code
+ *
+ * pipeline.apply(MongoDbIO.read()
+ *   .withUri("mongodb://localhost:27017")
+ *   .withDatabase("my-database")
+ *   .withCollection("my-collection")
+ *   // above three are required configuration, returns PCollection<String>
+ *
+ *   // rest of the settings are optional
+ *
+ * }</pre>
+ *
+ * <p>The source also accepts an optional configuration: {@code withFilter()} allows you to
+ * define a JSON filter to get subset of data.</p>
+ *
+ * <h3>Writing to MongoDB</h3>
+ *
+ * <p>MongoDB sink supports writing of Document (as JSON String) in a MongoDB.</p>
+ *
+ * <p>To configure a MongoDB sink, you must specify a connection {@code URI}, a {@code Database}
+ * name, a {@code Collection} name. For instance:</p>
+ *
+ * <pre>{@code
+ *
+ * pipeline
+ *   .apply(...)
+ *   .apply(MongoDbIO.write()
+ *     .withUri("mongodb://localhost:27017")
+ *     .withDatabase("my-database")
+ *     .withCollection("my-collection")
+ *     .withNumberSplit(30)
+ *
+ * }</pre>
+ */
+// TODO instead of JSON String, does it make sense to populate the PCollection with BSON Document or
+//  DBObject ??
+public class MongoDbIO {
+
+  public static Read read() {
+    return new Read();
+  }
+
+  public static Write write() {
+    return new Write();
+  }
+
+  private MongoDbIO() {}
+
+  /**
+   * A {@link PTransform} to read data from MongoDB.
+   */
+  public static class Read extends PTransform<PBegin, PCollection<String>> {
+
+    public Read withUri(String uri) {
+      return new Read(uri, database, collection, filter, numberSplit);
+    }
+
+    public Read withDatabase(String database) {
+      return new Read(uri, database, collection, filter, numberSplit);
+    }
+
+    public Read withCollection(String collection) {
+      return new Read(uri, database, collection, filter, numberSplit);
+    }
+
+    public Read withFilter(String filter) {
+      return new Read(uri, database, collection, filter, numberSplit);
+    }
+
+    public Read withNumberSplit(int numberSplit) {
+      return new Read(uri, database, collection, filter, numberSplit);
+    }
+
+    protected String uri;
+    protected String database;
+    protected String collection;
+    @Nullable
+    protected String filter;
+    protected int numberSplit;
+
+    private Read() {}
+
+    private Read(String uri, String database, String collection, String filter, int numberSplit) {
+      this.uri = uri;
+      this.database = database;
+      this.collection = collection;
+      this.filter = filter;
+      this.numberSplit = numberSplit;
+    }
+
+    @Override
+    public PCollection<String> apply(PBegin input) {
+      org.apache.beam.sdk.io.Read.Bounded bounded = org.apache.beam.sdk.io.
+          Read.from(createSource());
+      PTransform<PInput, PCollection<String>> transform = bounded;
+      return input.getPipeline().apply(transform);
```

Comment:
Yes, it works, but I had to specify the type of the `BoundedSource`:

```
private static class BoundedMongoDbSource extends BoundedSource<String>
```

instead of

```
private static class BoundedMongoDbSource extends BoundedSource
```